### PR TITLE
feat: add coveralls to ci

### PR DESCRIPTION
**Pull request by EvolveArt**
# Pull Request type

Please add the labels corresponding to the type of changes your PR introduces:

- Bugfix
- Feature
- Code style update (formatting, renaming)
- Refactoring (no functional changes, no API changes)
- Build-related changes
- Documentation content changes
- Testing
X Other (please describe): CI

## What is the current behavior?

Resolves: #NA

## What is the new behavior?

- Adds a new clippy check in CI
- Push coverage report to coveralls
- Spawns anvil from code in all tests, no need to spawn it on the side

## Does this introduce a breaking change?

No

## Other information

It was investigated to check number of unwraps but it doesn't make any sense to do in rust, it's fine to just make sure by ourselves there are no unsafe unwraps. We can always come back and add it where most unsafe ones have been removed.
As for the criterion benchmarking it should come in a separate PR as we first need to precisely know what should be benchmarked.

### CODE REVIEW

File: .github/workflows/coverage.yml [hunk 1]
```diff
@@ -61,3 +62,9 @@ jobs:
         with:
           GITHUB_TOKEN: ${{ github.token }}
           COVERAGE_FILE: coverage.cobertura.xml
+
+      - name: Coveralls
+        uses: coverallsapp/github-action@v2
+        with:
+          files: coverage.cobertura.xml
+          debug: true
```
Comment (antiyro) [inline .github/workflows/coverage.yml:51]: Could we add a coverage fail treshold here? something like: 60%?

Comment (EvolveArt) [reply .github/workflows/coverage.yml:51]: this is done automatically through coveralls UI

**Comment by antiyro** [comment]
could you clean CI for merge

**Comment by antiyro** [comment]
Actually I would take this PR to replace the comment from github actions with one from coveralls wdyt?

**Comment by EvolveArt** [comment]
> Actually I would take this PR to replace the comment from github actions with one from coveralls wdyt?

done

**Review by antiyro — APPROVED** [review]
lgtm

**Pull request merged**